test/h5test.h

```diff
@@ -82,6 +86,31 @@ H5TEST_DLLVAR char *paraprefix;
 H5TEST_DLLVAR MPI_Info h5_io_info_g; /* MPI INFO object for IO */
 #endif
 
+#define H5_API_TEST_FILENAME_MAX_LENGTH 1024
```

Comment:
Was there a reason for moving this and most of the below stuff into h5test.h? H5_api_test.h seems more appropriate for most of it.

Reply:
This is used by `generate_threadlocal_filename`, which is used by testframe to set up threadlocal information. The filename has to be generated by testframe since we can't know the number of executing threads before test runtime.

Reply:
Fair enough, though at that point this isn't related to the API tests, so a different name is probably warranted. `API_TEST_PASS`/`_FAIL`/`_ERROR` also seem like candidates for moving out of this header. Part of these problems would go away with the remaining test framework refactoring changes merged since we'd have an actual testframe.h header at that point.